frontend/packages/pipelines-plugin/src/components/pipelineruns/PipelineRunsResourceList.tsx

```diff
@@ -17,7 +21,31 @@ const PipelineRunsResourceList: React.FC<
     PipelineRunsResourceListProps
 > = (props) => {
   const { t } = useTranslation();
-  const badge = usePipelineTechPreviewBadge(props.namespace);
+  const ns = props.namespace || props?.match.params.ns;
+  const badge = usePipelineTechPreviewBadge(ns);
+  const [TRPlrs, TRPlrsLoaded, TRPlrsLoadError] = useTRPipelineRuns(ns);
+  const [k8sPlrs, k8sPlrsLoaded, k8sPlrsLoadError] = useK8sWatchResource<PipelineRunKind[]>({
+    isList: true,
+    kind: referenceForModel(PipelineRunModel),
+    namespace: ns,
+  });
+  const mergedPlrs =
+    TRPlrsLoaded || (k8sPlrsLoaded && !k8sPlrsLoadError)
+      ? k8sPlrs.concat(
+          TRPlrs.filter(
+            (plr) => !k8sPlrs.some((k8sPlr) => k8sPlr.metadata.uid === plr.metadata.uid),
+          ),
```

Comment:
This will iterate over all k8s resources, for each result-PLR. Please extract the k8s uid first into a `Set` and remove the duplicates then with [set.has](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/has)